public/app/plugins/datasource/cloud-monitoring/components/VisualMetricQueryEditor.test.tsx

```diff
@@ -69,30 +221,66 @@ describe('VisualMetricQueryEditor', () => {
     });
     const defaultQuery = { ...query, ...defaultTimeSeriesList(datasource), filters: query.filters };
 
-    render(
-      <VisualMetricQueryEditor
-        refId="refId"
-        query={query}
-        datasource={datasource}
-        onChange={onChange}
-        customMetaData={{}}
-        variableOptionGroup={{ options: [] }}
-        aliasBy={''}
-        onChangeAliasBy={jest.fn()}
-      />
-    );
+    render(<VisualMetricQueryEditor {...defaultProps} onChange={onChange} datasource={datasource} query={query} />);
     expect(document.body).toHaveTextContent('metric.test_label');
-    expect(await screen.findByText('Delta')).not.toBe(null);
-    expect(await screen.findByText('metric.test_groupby')).not.toBe(null);
+    expect(await screen.findByText('Delta')).toBeInTheDocument();
+    expect(await screen.findByText('metric.test_groupby')).toBeInTheDocument();
     const metric = await screen.findByLabelText('Metric name');
-    await openMenu(metric);
+    openMenu(metric);
     await select(metric, 'metricName2', { container: document.body });
     expect(onChange).toBeCalledWith(
       expect.objectContaining({ filters: ['metric.test_label', '=', 'test', 'AND', 'metric.type', '=', 'type2'] })
     );
     expect(query).toEqual(defaultQuery);
     expect(document.body).toHaveTextContent('metric.test_label');
-    expect(await screen.queryByText('Delta')).toBe(null);
-    expect(await screen.queryByText('metric.test_groupby')).toBe(null);
+    expect(await screen.queryByText('Delta')).not.toBeInTheDocument();
+    expect(await screen.queryByText('metric.test_groupby')).not.toBeInTheDocument();
+  });
+
+  it('updates labels on time range change', async () => {
```

Comment:
:oof: tricky test :sweat_smile:

Reply:
🤣  I finally got it working and I'm very happy about this